lib/main.rb

```diff
@@ -0,0 +1,141 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+
+# my_Enumerable
+module Enumerable
+  # -------------my_each-----------
+
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    self_item = self
+    count = 0
+    items = to_a if self_item.class == Hash || Range
+
+    until count > items.length - 1
+      yield(items[count])
+      count += 1
+    end
+    self
+  end
+
+  # ------------my_each_with_index----------
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    self_item = self
+    items = to_a if self_item.class == Hash || Range
+    count = 0
+    until count > items.length - 1
+      yield(items[count], count)
+      count += 1
+    end
+    self
+  end
+
+  # ------------my_select----------
+
+  def my_select
+    return to_enum(:my_each) unless block_given?
+
+    items = []
+    my_each { |item| items << item if yield item }
+    items
+  end
+
+  # ------------my_all?----------
+
+  def my_all?(argument = nil)
+    if block_given?
+      my_each { |item| return false unless yield(item) }
+      return true
+    end
+    argument.nil? ? argument.class.to_s : my_all? { |item| item }
+
+    if argument.class.to_s == 'Class'
+      my_all? { |item| item.is_a? argument }
+    elsif argument.class.to_s == 'Regexp'
+      my_all? { |item| item =~ argument }
+    elsif argument.nil?
+      my_each { |item| return false if item == false || item.nil? }
+    else
+      my_all? { |item| item == argument if item != true }
+    end
+  end
```

Comment:
when no block or argument is given it should return true when none of the collection members are false or nil. Kindly check below.
### YOURS
`true_array = [1, true, 'hi', []]
puts true_array.my_all?
1
true
hi
`
### ORIGINAL
`true_array = [1, true, 'hi', []]
puts true_array.all?
true
`